ams/AssetAnalyzer.cs

```diff
@@ -132,63 +134,109 @@ await foreach (var locator in locators!)
                                                         _analysisOptions.CreationTimeEnd);
 
-            var reportGenerator = new ReportGenerator(_globalOptions.HtmlReportFile, _globalOptions.JsonReportFile, _logger);
-            reportGenerator.WriteHeader();
-
-            await _resourceProvider.SetStorageResourceGroupsAsync(account, cancellationToken);
-            var assets = account.GetMediaAssets()
-                .GetAllAsync(resourceFilter, cancellationToken: cancellationToken);
             var statistics = new AssetStats();
             var assetTypes = new ConcurrentDictionary<string, int>();
 
-            List<MediaAssetResource>? filteredList = null;
-
-            if (resourceFilter != null)
+            if (_analysisOptions.IsStorageAcc)
             {
-                // When a filter is used, it usually include a small list of assets,
-                // The total count of asset can be extracted in advance without much perf hit.
-                filteredList = await assets.ToListAsync(cancellationToken);
+                var (storageClient, accountId) = await _resourceProvider.GetStorageAccount(_analysisOptions.AccountName, cancellationToken);
 
-                totalAssets = filteredList.Count;
-            }
+                double totalItems = await GetStorageBlobMetricAsync(accountId, cancellationToken);
+                var containers = storageClient.GetBlobContainersAsync(
+                              prefix: _analysisOptions.Prefix, cancellationToken: cancellationToken);
+                _logger.LogInformation("The total contianers count of the storage account is {count}.", totalItems);
+                List<BlobContainerItem>? filteredList = null;
 
-            _logger.LogInformation("The total assets to handle in this run is {count}.", totalAssets);
+                if (_analysisOptions.Prefix != null)
+                {
+                    filteredList = await containers.ToListAsync();
+                    totalItems = filteredList.Count;
+                }
+                _logger.LogInformation("The total containers to handle in this run is {count}.", totalItems);
+                var channel = Channel.CreateBounded<double>(1);
+                var progress = ShowProgressAsync("Analyzing Containers", "Assets", totalItems, channel.Reader, cancellationToken);
+                var writer = channel.Writer;
+                await MigrateInParallel(containers, filteredList, async (container, cancellationToken) =>
+                {
+                    //  var storage = await _resourceProvider.GetStorageAccountAsync(account, asset, cancellationToken);
+                    var result = await AnalyzeAsync(container, storageClient, cancellationToken);
+                    var assetType = result.AssetType ?? "unknown";
+                    assetTypes.AddOrUpdate(assetType, 1, (key, value) => Interlocked.Increment(ref value));
+                    reportGenerator?.WriteRecord(result);
```

Comment:
The reportGenerator is never null after you sync the source code from Main branch.  so no ? part is needed.